pkg/cli/clusteroperator/restartkubelet/restart_kubelet.go

```diff
@@ -0,0 +1,229 @@
+package restartkubelet
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/oc/pkg/helpers/conditions"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+)
+
+var (
+	nodeKind = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Node"}
+
+	//go:embed restart-namespace.yaml
+	namespaceYaml []byte
+	namespace     = resourceread.ReadNamespaceV1OrDie(namespaceYaml)
+
+	//go:embed restart-pod-template.yaml
+	podYaml []byte
+	pod     = resourceread.ReadPodV1OrDie(podYaml)
+)
+
+type RestartKubeletRuntime struct {
+	ResourceFinder genericclioptions.ResourceFinder
+	KubeClient     kubernetes.Interface
+
+	DryRun bool
+
+	ImagePullSpec            string
+	NumberOfNodesInParallel  int
+	PercentOfNodesInParallel int
+
+	Printer printers.ResourcePrinter
+	genericclioptions.IOStreams
+}
+
+func (o *RestartKubeletRuntime) Run(ctx context.Context) error {
+	interestingNodes, err := o.GetInterestingNodes(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to get interesting nodes: %w", err)
+	}
+
+	numberOfNodesInParallel := 1
+	switch {
+	case o.NumberOfNodesInParallel > 0:
+		numberOfNodesInParallel = o.NumberOfNodesInParallel
+	case o.PercentOfNodesInParallel > 0:
+		numberOfNodesInParallel = len(interestingNodes) / o.PercentOfNodesInParallel
+	}
+	if numberOfNodesInParallel < 1 {
+		numberOfNodesInParallel = 1
+	}
+
+	// create a namespace to work in
+	nsName := "!!-dry-run"
```

Comment:
Hmm...do we need a full new namespace for this?  Why not `kube-system` or something?

Reply:
kube-system is naughty to use.  Shared namespaces are fraught.  Notice the cleanup on this script is a clean namespace delete to avoid cruft.